app/js/elements/fxos-inspector.js

```diff
@@ -0,0 +1,485 @@
+    text-decoration: inherit;
+    text-transform: none;
+    text-rendering: optimizeLegibility;
+    font-size: 30px;
+    -webkit-font-smoothing: antialiased;
+  }
+  [flex] {
+    display: flex;
+    justify-content: space-between;
+    padding: 0 5px;
+  }
+  [flex] > * {
+    flex: 1 1 auto;
+    margin-right: 5px;
+    margin-left: 5px;
+  }
+  #container {
+    box-sizing: border-box;
+    position: relative;
+    width: 100%;
+    height: 100%;
+    overflow-x: hidden;
+    overflow-y: auto;
+  }
+  gaia-list a {
+    position: relative;
+    text-decoration: none;
+  }
+  gaia-list [hidden] {
+    display: none;
+  }
+  gaia-list h3 {
+    max-width: calc(100% - 2em);
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
+  gaia-list [data-icon] {
+    position: absolute;
+    padding: 1.6em 1em;
+    top: 0;
+    right: 0;
+  }
+  gaia-list .value {
+    color: #00aac5;
+  }
+  fxos-code-editor {
+    display: block;
+    width: 100%;
+    height: calc(100% - 100px);
+  }
+</style>
+<div id="container"></div>
+<gaia-dialog-prompt id="new-attribute-name">
+  Enter new attribute name
+</gaia-dialog-prompt>`;
+
+proto.createdCallback = function() {
+  this.shadow = this.createShadowRoot();
+  this.shadow.innerHTML = template;
+
+  this.container = this.shadow.querySelector('#container');
+  this.newAttributeName = this.shadow.querySelector('#new-attribute-name');
+
+  this.keyPath = [];
+
+  // Handle all clicks within this component.
+  this.container.addEventListener('click', (evt) => {
+    // Handle case where <a href> was clicked.
+    var target = evt.target.closest('a');
+    if (target) {
+      evt.preventDefault();
+
+      // Handle case where <i data-action="remove"> was clicked.
+      if (evt.target.dataset.action === 'remove') {
+        this._handleRemoveAttributeClick(target);
+      }
+
+      // Otherwise, treat as regular list item click.
+      else {
+        this._handleListItemClick(target);
+      }
+      return;
+    }
+
+    // Handle case where <gaia-button> was clicked.
+    target = evt.target.closest('gaia-button');
+    if (target) {
+      evt.preventDefault();
+      this._handleButtonClick(target);
+      return;
+    }
+  });
+
+  // Filter the list of properties 500ms after a keypress
+  // in the <gaia-text-input> search box.
+  this.container.addEventListener('keyup', (evt) => {
+    var textInput = evt.target.closest('gaia-text-input');
+    if (!textInput) {
+      return;
+    }
+
+    clearTimeout(this._searchTimeout);
+    this._searchTimeout = setTimeout(() => {
+      var value = textInput.value.toLowerCase();
+      var items = [].slice.apply(this.container.querySelectorAll('gaia-list a[data-search]'));
+
+      // Toggle each list items' [hidden] attribute if the search text
+      // isn't blank and the items' [data-search] value contains the
+      // current search text.
+      items.forEach(i => i.hidden = value && i.dataset.search.indexOf(value) === -1);
```

Comment:
I saw your previous comment about this and that's why I added the comment above it. Its very concise and I can't think of a way to break it up to make it much clearer because we need to set the `hidden` property to either `true` or `false` for every single item.